File: leetcode/Array/0121_Best_Time_to_Buy_and_Sell_Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        N = len(prices)
        slow, fast = 0, 1
        max_profit = 0
        while fast < N:
            if prices[fast] > prices[slow]:
                max_profit = max(max_profit, prices[fast] - prices[slow])
                fast += 1
            else:
                slow = fast
                fast += 1

        return max_profit

File: leetcode/Array/0121_Best_Time_to_Buy_and_Sell_Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() <= 1)
            return 0;
        int minLocal = prices[0], result = 0;
        for (int i = 0; i < prices.size(); i++) {
            if (prices[i] < minLocal)
                minLocal = prices[i];
            if (prices[i] - minLocal > result)
                result = prices[i] - minLocal;
        }
        return result;
    }
};